src/main/java/athleticli/data/sleep/Sleep.java

```diff
@@ -1,35 +1,118 @@
 package athleticli.data.sleep;
 
+import java.time.Duration;
+import java.time.LocalDate;
 import java.time.LocalDateTime;
+import java.time.LocalTime;
 import java.time.format.DateTimeFormatter;
+import java.util.Locale;
 
 /**
  * Represents a sleep record.
  */
 public class Sleep {
 
-    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd-MM-YYYY HH:mm");
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("MMMM d, " +
+            "yyyy 'at' h:mm a", Locale.ENGLISH);
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern(
+            "yyyy-MM-dd", Locale.ENGLISH);
 
-    private LocalDateTime from;
-    private LocalDateTime to;
+    private final LocalDateTime startDateTime;
+    private final LocalDateTime toDateTime;
+
+    private LocalTime sleepingDuration;
+
+    private final LocalDate sleepDate;
 
     /**
-     * Constructor for Sleep.
-     *
-     * @param from Start time of the sleep.
-     * @param to End time of the sleep.
+     * Generates a new sleep record with some basic stats.
+     * 
+     * @param startDateTime Start time of the sleep.
+     * @param toDateTime    End time of the sleep.
      */
-    public Sleep(LocalDateTime from, LocalDateTime to) {
-        this.from = from;
-        this.to = to;
+    public Sleep(LocalDateTime startDateTime, LocalDateTime toDateTime) {
+        this.startDateTime = startDateTime;
+        this.toDateTime = toDateTime;
+        this.sleepingDuration = calculateSleepingDuration();
+        this.sleepDate = calculateSleepDate();
+    }
+
+    public LocalDateTime getStartDateTime() {
+        return startDateTime;
+    }
+
+    public LocalDateTime getToDateTime() {
+        return toDateTime;
+    }
+
+    public LocalDate getSleepDate() {
+        return sleepDate;
+    }
+
+    public LocalTime getSleepingTime() {
+        return sleepingDuration;
     }
 
     /**
-     * toString method for Sleep.
+     * Calculate the sleeping duration based on start and end time.
+     * Factor in the possibility of sleeping past midnight.
      *
+     * @return sleeping duration.
+     */
+    private LocalTime calculateSleepingDuration() {
+        Duration duration = Duration.between(startDateTime, toDateTime);
+        long seconds = duration.getSeconds();
+        return LocalTime.ofSecondOfDay(seconds);
+    }
+
+    /**
+     * Calculate the sleep date based on start time.
+     * Factor in the possibility of sleeping past midnight.
+     * We are assuming that the user sleeps before 6am even if the user sleeps past midnight.
+     * @return sleep date.
+     */
+    private LocalDate calculateSleepDate() {
+        if (startDateTime.getHour() < 6) {
```

Comment:
It takes good care of the sleeping habits of actual users!

Reply:
Thanks!